torch/csrc/api/include/torch/nn/init.h

```diff
@@ -1,27 +1,32 @@
 #pragma once
 
+#include <c10/util/variant.h>
 #include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/enum.h>
 #include <torch/types.h>
 
 namespace torch {
 namespace nn {
 namespace init {
 
-enum class Nonlinearity {
-  Linear,
-  Conv1D,
-  Conv2D,
-  Conv3D,
-  ConvTranspose1D,
-  ConvTranspose2D,
-  ConvTranspose3D,
-  Sigmoid,
-  Tanh,
-  ReLU,
-  LeakyReLU
-};
-
-enum class FanMode { FanIn, FanOut };
+using Nonlinearity = c10::variant<
```

Comment:
@yf225 This PR looks good. Just one comment: Do you think that it's good idea to declare the variants where they are used (and not at the top of the file)?

Reply:
@ShahriarSS Thanks a lot for reviewing! I think because the variant type `Nonlinearity` is used in multiple places, it would be better if we declare it in a common place first (e.g. at the top of the file).